scripts/build-docs.ts

```diff
@@ -0,0 +1,126 @@
+import * as fs from "fs-extra";
+import { Application, TSConfigReader, TypeDocOptions } from "typedoc";
+
+const OUTPUT_DIR = "dist/docs";
+const INDEX_FILE = "doc/index.md";
+
+const GITHUB_FORMAT = 'github';
+const PC_DEV_CENTER_FORMAT = 'purecloudDevCenter';
+const SUPPORTED_DOC_OUTPUT_FORMATS = [GITHUB_FORMAT, PC_DEV_CENTER_FORMAT];
+const docMdOutputFormat = process.env.DOC_MD_OUTPUT_FORMAT || PC_DEV_CENTER_FORMAT;
+if (SUPPORTED_DOC_OUTPUT_FORMATS.indexOf(docMdOutputFormat) < 0) {
+    console.error(`Unknown MD Output Format Specified: '${docMdOutputFormat}'`);
+    process.exit(1);
+}
+
+let flagDocsAsPreview = true;
+if (process.env.FLAG_DOCS_AS_PREVIEW) {
+    ['false', 'f', '0'].forEach(currFalseValue => {
+        if (process.env.FLAG_DOCS_AS_PREVIEW === currFalseValue) {
+            flagDocsAsPreview = false;
+        }
+    });
+}
+
+// Options added by the Typedoc Markdown Plugin
+interface MarkdownPluginOptions {
+    skipSidebar: boolean;
+    hideBreadcrumbs: boolean;
+}
+
+type DocsConfig = Partial<TypeDocOptions & MarkdownPluginOptions>;
+
+const docsConfig: DocsConfig = {
+    mode: "file",
+    name: "Client App SDK",
+    logger: 'none',
+    theme: "docusaurus", // Theme that generates output closest to Dev Center format
+    skipSidebar: true, // Disable sidebar generation
+    disableSources: true, // Don't show the file for each method/class
+    hideBreadcrumbs: true, // Dev Center handles breadcrumbs for us
+    readme: "none",
+    plugin: [
+        'typedoc-plugin-markdown',
+        'typedoc-plugin-no-inherit'
+    ],
+    exclude: [
+        "**/utils/**",
+        "**/.babelrc",
+        "**/.eslintrc.js",
+        "**/*+(Spec).ts"
+    ],
+    excludeExternals: true,
+    excludePrivate: true,
+    excludeProtected: true,
+    excludeNotExported: true,
+    excludeNotDocumented: true
+};
+
+const transformLinks = (buffer: string, ext: string) => {
+    // Regex to replace the following patterns (ext = "html"):
+    // [link1](api.md) -> [link1](api.html)
+    // [link2](api.md#someref) -> [link2](api.html#someref)
+    return buffer.replace(/(\[[^\]]+\][^)]+)(\.md)(\)|#[^)]*\))/gm, `$1${ext}$3`);
+};
+
+const prependDocHeaderAttribute = (buffer: string, attr: string) => {
+    if (!buffer.startsWith('---')) {
+        throw new Error('Doc header symbol "---" not found');
+    }
+    return buffer = `---\n${attr}${buffer.substring('---'.length)}`;
+};
+
+(async () => {
+    const app = new Application();
+
+    // So TypeDoc can load tsconfig.json
+    app.options.addReader(new TSConfigReader());
+
+    app.bootstrap(docsConfig);
+
+    const project = app.convert(app.expandInputFiles(["src"]));
+    if (!project) throw new Error("Unable to generate typedoc project");
+
+    // Rendered docs
+    app.generateDocs(project, OUTPUT_DIR);
+
+    // Flatten typedoc's generated classes folder
+    const classesDir = path.join(OUTPUT_DIR, "classes");
+    const fileNames = await fs.readdir(classesDir);
```

Comment:
Will classes always be a flat dir of files?  Just want to make sure a dir wouldn't get copied to dist/docs and then fail on 113 below.

I dont think it's necessary to add a file type guard here at this time if it's always flat though; the build will break and we'll see that.  Just wanted to make sure that was the case now and there weren't any edge cases that would blow us up in the short-term.

Reply:
Yep, from what I've seen it looks like `classes` is always just a dir of files. The only edge case with our typedoc docs build is if someone went in and added an interface that is both exported and documented then another folder will be generated called `interfaces` alongside `classes`. The docs build will fail in this case as we'll fail when we encounter line 113, but it wouldn't hurt to add something to the README about this and maybe a more specific error.